posthog/api/prompt.py

```diff
@@ -6,7 +7,79 @@
 from rest_framework.response import Response
 
 from posthog.api.routing import StructuredViewSetMixin
-from posthog.models.prompt import UserPromptSequenceState, get_active_prompt_sequences
+from posthog.client import query_with_columns
+from posthog.models.prompt.constants import prompts_config
+from posthog.models.prompt.prompt import UserPromptSequenceState
+
+SELECT_PROMPT_TRIGGER_SQL = """
+SELECT
+    event,
+    properties,
+    timestamp
+FROM events
+WHERE
+team_id = %(team_id)s
+AND event = 'prompt trigger'
+AND events.distinct_id in [%(distinct_id)s]
```

Comment:
Since it's a single distinct ID, why not do this: 
```suggestion
AND events.distinct_id = %(distinct_id)s
```

and if they're multiple, add it to the `distinct_id` parameter, and use it here like so:
`events.distinct_id in %(distinct_id)s`

---

Also note that doing this in clickhouse is relatively expensive, if we do this for every user coming in (CH doesn't do well with relational type queries, like getting events for a single distinct_id)